src/disco/reducers/discoResults.js

```diff
@@ -1,11 +1,95 @@
-import { LOAD_DISCO_RESULTS } from 'disco/constants';
+/* @flow */
+import invariant from 'invariant';
 
-export default function discoResults(state = [], { type, payload }) {
-  switch (type) {
+import type { AddonType, ExternalAddonType } from 'core/types/addons';
+
+export const GET_DISCO_RESULTS: 'GET_DISCO_RESULTS' = 'GET_DISCO_RESULTS';
+export const LOAD_DISCO_RESULTS: 'LOAD_DISCO_RESULTS' = 'LOAD_DISCO_RESULTS';
+
+export type ExternalDiscoAddonMap = {
+  [guid: $PropertyType<AddonType, 'guid'>]: ExternalAddonType,
+};
+
+type ExternalDiscoResultType = {|
+  // normalizr injects the add-on's GUID in the `addon` prop.
+  addon: $PropertyType<AddonType, 'guid'>,
+  description: string | null,
+  heading: string,
+  is_recommendation: boolean,
+|};
+
+export type ExternalDiscoResultsType = {|
+  entities: {|
+    addons: ExternalDiscoAddonMap,
+    discoResults: {| [guid: string]: ExternalDiscoResultType |},
+  |},
+  result: {|
+    count: number,
+    results: Array<$PropertyType<AddonType, 'guid'>>,
+  |},
+|};
+
+type DiscoResultsState = Array<ExternalDiscoResultType>;
+
+export const initialState: DiscoResultsState = [];
+
+type GetDiscoResultsParams = {|
+  errorHandlerId: string,
+  taarParams: {
+    [name: string]: string,
+    platform: string,
+  },
+|};
+
+export type GetDiscoResultsAction = {|
+  type: typeof GET_DISCO_RESULTS,
+  payload: GetDiscoResultsParams,
+|};
+
+export function getDiscoResults({
+  errorHandlerId,
+  taarParams,
+}: GetDiscoResultsParams = {}): GetDiscoResultsAction {
+  invariant(errorHandlerId, 'errorHandlerId is required');
+  invariant(taarParams.platform, 'taarParams.platform is required');
+
+  return {
+    type: GET_DISCO_RESULTS,
+    payload: { errorHandlerId, taarParams },
+  };
+}
+
+type LoadDiscoResultsParams = ExternalDiscoResultsType;
+
+type LoadDiscoResultsAction = {|
+  type: typeof LOAD_DISCO_RESULTS,
+  payload: LoadDiscoResultsParams,
+|};
+
+export function loadDiscoResults({
+  entities,
+  result,
+}: LoadDiscoResultsParams = {}): LoadDiscoResultsAction {
+  invariant(entities, 'entities are required');
+  invariant(result, 'result is required');
+
+  return {
+    type: LOAD_DISCO_RESULTS,
+    payload: { entities, result },
+  };
+}
+
+type Action = LoadDiscoResultsAction | GetDiscoResultsAction;
+
+export default function discoResults(
```

Comment:
can we call this reducer? it seems that we are doing this in other places...

Reply:
I tend to agree with you but it is not consistent.. there are reducers that are named.

Reply:
ok I just thought our eventual goal was [this](https://github.com/mozilla/addons-frontend/issues/1299) so I figured why you were changing this, we should update:)

I guess this is still moving in this direction tho so 👏

Reply:
Yeah, naming it `reducer` or not does not really matter. It is now a Duck(s) :)

Reply:
ok cool, I wasn't sure if that mattered..cool tho now I know :)